http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/service/RuntimeServiceInterp.scala

```diff
@@ -281,6 +293,46 @@ class RuntimeServiceInterp[F[_]: Parallel](blocker: Blocker,
       _ <- publisherQueue.enqueue1(StartRuntimeMessage(runtime.id, Some(ctx.traceId)))
     } yield ()
 
+  override def updateRuntime(
+    userInfo: UserInfo,
+    googleProject: GoogleProject,
+    runtimeName: RuntimeName,
+    req: UpdateRuntimeRequest
+  )(implicit as: ApplicativeAsk[F, RuntimeServiceContext]): F[Unit] =
+    for {
+      // throw 404 if not existent
+      runtimeOpt <- clusterQuery.getActiveClusterByNameMinimal(googleProject, runtimeName).transaction
+      runtime <- runtimeOpt.fold(F.raiseError[Runtime](RuntimeNotFoundException(googleProject, runtimeName)))(F.pure)
+      // throw 404 if no GetClusterStatus permission
+      hasPermission <- authProvider.hasNotebookClusterPermission(runtime.internalId,
```

Comment:
should this be 403?

Reply:
Our pattern is to 404 (e.g. pretend the runtime doesn't exist) if the user doesn't have `GetClusterStatus` permission. We return 403 if the user can _view_ the runtime but can't perform some other action.

Reply:
gotcha, that's very helpful...can you add the comment in code as well :)